app/src/main/java/org/mozilla/focus/history/HistoryItemAdapter.java

```diff
@@ -121,15 +125,17 @@ public void onResourceReady(Bitmap resource, Transition<? super Bitmap> transiti
                 }
 
                 final PopupMenu popupMenu = new PopupMenu(mContext, siteVH.btnMore);
-                popupMenu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
-                    @Override
-                    public boolean onMenuItemClick(MenuItem menuItem) {
-                        if (menuItem.getItemId() == R.id.browsing_history_menu_delete) {
-                            BrowsingHistoryManager.getInstance().delete(item.getId(), HistoryItemAdapter.this);
-                            TelemetryWrapper.historyRemoveLink();
+                popupMenu.setOnMenuItemClickListener(menuItem -> {
+                    if (menuItem.getItemId() == R.id.browsing_history_menu_delete) {
+                        try {
+                            new FileUtils.DeleteFileThread(new File(new URI(item.getFavIconUri()))).start();
+                        } catch (URISyntaxException e) {
```

Comment:
How about let's separate the try/catch exception from new DeleteFileThread.
Can we reduce the usage of creating a new thread for each action?